chainer/functions/accuracy.py

```diff
@@ -2,9 +2,23 @@
 
 from chainer import cuda
 from chainer import function
+from chainer.utils import type_check
 
 
 class Accuracy(function.Function):
+
+    def check_type_forward(self, in_types):
+        type_check.expect(in_types.size() == 2)
+        x_type, t_type = in_types
+
+        type_check.expect(
+            x_type.dtype == numpy.float32,
+            x_type.ndim == 2,
```

Comment:
`ndim` of `x_type` need not to be 2. It can have trailing 1's which starts from third dimension (like (10, 3, 1, 1, 1 ...)).